src/Features/CSharp/Portable/UnsafeProject/CSharpUnsafeProjectCodeFixProvider.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.UpgradeProject;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnsafeProject
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class CSharpUnsafeProjectCodeFixProvider : CodeFixProvider
+    {
+        private const string CS0227 = nameof(CS0227); // error CS0227: Unsafe code may only appear if compiling with /unsafe
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(CS0227);
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            // Fix all could be implemented, but doesn't seem as important because for this to be useful,
```

Comment:
i agree.  i do not think we need fix all.  i think we can just state that we're ok with users having to explicitly opt in each project, and we don't have to specify how to do fix all at all.

Reply:
I agree we shouldn't offer FixAll. Maybe rephrase the comment to that effect ("We don't offer FixAll because unsafe code is by itself an uncommon case and it is unlikely the user wants to convert multiple projects to use unsafe code at once.")

---
In reply to: [178472129](https://github.com/dotnet/roslyn/pull/25875#discussion_r178472129) [](ancestors = 178472129)

Reply:
#resolved